hackathon_site/hardware/views.py

```diff
@@ -86,3 +89,14 @@ def post(self, request, *args, **kwargs):
             return HttpResponseServerError()
         response_data = response_serializer.data
         return Response(response_data, status=status.HTTP_201_CREATED)
+
+    @transaction.atomic
```

Comment:
Does this need to be atomic? This has an impact on performance, so we don't want to do it unless there's a race condition (which I don't think there is)